examples-nosql-java-sdk/CreateLoadComplexTable/README.md

```diff
@@ -0,0 +1,347 @@
+
+  Specifies the profile in the credentials file from which to retrieve the credentials to use when authenticating with the Oracle NoSQL Database Cloud Service. If this entry is not specified, then the default profile will be used.
+
+## Overriding Configuration Entries
+
+  For convenience, each entry in the configuration file can be overridden on the command line by setting a system property with name equal to the name of the configuration entry. For example, suppose your environment is the OCI Command Shell and you have specified the configuration file `~/.oci/my-config.json` with contents like, 
+````
+{
+    "credentialsType" : "token",
+    "region" : "us-sanjose-1",
+    "compartment" : "<ocid.of.compartment.containing.table>", 
+    "table" : "my_table_1",
+    "readUnits" : "50",
+    "writeUnits" : "50",
+    "storageGb" : "25"
+}
+````
+  Specifying the configuration above when executing the following command will cause the program to authenticate using the OCI Command Shell delegation token when creating and populating a table named `my_table_1`; where the table will have 10 rows (the default), and will be allocated 50 read units, 50 write units, and maximum storage space of 25 Gbs.
+
+````
+java -jar ./lib/complextable-1.0.0.jar -config ~/.oci/my-config.json
+````
+
+  Next, suppose you wish to create a new table named "my_table_2" with 51 rows. And suppose you wish to allocate to that new table, 10 read units, 15 write units, and 3 Gbs maximum storage. To do this, you can create a new configuration file with new values, or you can use the configuration file above, but override the deired configuration entries on the command line. For example,
+
+````
+java -Dtable=my_table_2 -DreadUnits=10 -DwriteUnits=15 -DstorageGb=3 \
+     -jar ./lib/complextable-1.0.0.jar -config ~/.oci/my-config.json
+````
+
+  This feature may be useful when testing or writing scripts that run the utility at regular intervals; using a single configuration file 'template', but overriding the entries you wish to manipulate. 
+
+## Configuring Logging
+
+  The system properties that configure the loggers used during execution are optional. If those system properties are not specified, then the utility will produce no logging output.
+
+  The CreateLoadComplexTable example program executes software from different third-party libraries, where each library defines its own set of loggers with different namespaces. For convenience, this example provides two logging configuration files as part of the release; one to configure logging mechanisms based on java.util.logging, and one for loggers based on Log4j2.
+
+  By default, the logger configuration files provided with the utility are designed to produce minimal output as the program executes. If you wish to see more verbose output from the various components employed during execution, then you should increase the logging levels of the specific loggers referenced in the appropriate logging configuration file.
+
```

Comment:
# Troubleshooting

Be sure to meet the prerequisites. Otherwise, you can experience the following errors:

1. Not using the minimum Java version - `Java 11 or higher`

    **Action**: use Java 11 or higher.
    **Note**: Use the command `java -version` to validate the version used.

    ```
    [ERROR] COMPILATION ERROR : 
    [INFO] -------------------------------------------------------------
    [ERROR] javac: invalid flag: --release
    Usage: javac <options> <source files>
    use -help for a list of possible options
    ...
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project complextable: Compilation failure
    `````

2) JAVA_HOME environment variable is not correctly set
 
    **Action**: set the JAVA_HOME correctly.
    **Note**: as a workaround, you can also use the following maven parameter `-Dmaven.javadoc.skip=true`

    ````
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (javadoc-jar) on project complextable: MavenReportException: Error while generating Javadoc: Unable to find javadoc command: The environment variable JAVA_HOME is not correctly set. -> [Help 1]`
    ````